operator-common/src/main/java/io/strimzi/operator/common/AdminClientProvider.java

```diff
@@ -26,7 +26,17 @@ public interface AdminClientProvider {
     Admin createAdminClient(String bootstrapHostnames, PemTrustSet kafkaCaTrustSet, PemAuthIdentity authIdentity);
 
     /**
-     * Create a Kafka Admin interface instance
+     * Create a Kafka Admin interface instance for controllers
+     *
+     * @param controllerBootstrapHostnames Kafka controller hostname to connect to for administration operations
+     * @param kafkaCaTrustSet Trust set for connecting to Kafka
+     * @param authIdentity Identity for TLS client authentication for connecting to Kafka
+     * @return Instance of Kafka Admin interface
+     */
+    Admin createControllerAdminClient(String controllerBootstrapHostnames, PemTrustSet kafkaCaTrustSet, PemAuthIdentity authIdentity);
```

Comment:
I didn't get why we need new methods for the admin client provider if the creation of an admin client for brokers or controllers differs just for the bootstrap parameter value. Also the corresponding implementation looks exactly the same comparing `createAdminClient` and `createControllerAdminClient`. What am I missing here?

Reply:
Ah right there is a different admin client config to set. Let's take the separated methods.